control/statesp.py

```diff
@@ -612,14 +612,20 @@ def dcgain(self):
             at the origin
         """
         try:
-            gain = np.asarray(self.D -
-                              self.C.dot(np.linalg.solve(self.A, self.B)))
+            if self.isctime():
+                gain = np.asarray(self.D -
+                                    self.C.dot(np.linalg.solve(self.A, self.B)))
+            else:
+                gain = self.horner(1)
         except LinAlgError:
-            # zero eigenvalue: singular matrix
-            return np.nan
+            # eigenvalue at DC
+            # TODO: should this be of shape (outputs,inputs) ?
+            # TODO: inf rather than nan?
+            # TODO: could one not do a bit better, maybe by finding a least squares solution?
```

Comment:
Fair enough.  On reflection, I think the shape remark is correct, and the other two require discussion.  Shall I change this to `np.tile(np.nan,(self.outputs,self.inputs))` (with unit test), and open an issue on whether we can do better?  FWIW, after a bit of thought and some experimentation, I'm not sure how to do better than evaluating DC gain on a `minreal`ed version of each input-output pair.